packages/react-reconciler/src/ReactChildFiber.js

```diff
@@ -848,28 +853,32 @@ function createChildReconciler(
           return null;
         }
 
-        return updateFragment(
+        const prevDebugInfo = pushDebugInfo(newChild._debugInfo);
+        const updated = updateFragment(
           returnFiber,
           oldFiber,
           newChild,
           lanes,
           null,
-          mergeDebugInfo(debugInfo, newChild._debugInfo),
         );
+        currentDebugInfo = prevDebugInfo;
+        return updated;
       }
 
       // Usable node types
       //
       // Unwrap the inner value and recursively call this function again.
       if (typeof newChild.then === 'function') {
         const thenable: Thenable<any> = (newChild: any);
-        return updateSlot(
+        const prevDebugInfo = pushDebugInfo((thenable: any)._debugInfo);
+        const updated = updateSlot(
           returnFiber,
           oldFiber,
           unwrapThenable(thenable),
           lanes,
-          debugInfo,
         );
```

Comment:
I thought we need to `pushDebugInfo` where we used to `mergeDebugInfo` and don't need to `pushDebugInfo` when we just passed `debugInfo` as an argument. But here we're now pushing where we didn't used to merge. Is this a bug fix or change in behavior?

Reply:
It's a bug fix. I missed this before. It should've merged.